cmake/ElektraCache.cmake

```diff
@@ -346,11 +346,16 @@ set (LIB_SUFFIX ""
 		"Optional suffix to use on lib folders (e.g. 64 for lib64)"
     )
 
-set(MEMORYCHECK_SUPPRESSIONS_FILE
+set (MEMORYCHECK_SUPPRESSIONS_FILE
 		"${CMAKE_SOURCE_DIR}/tests/valgrind.suppression"
 		CACHE FILEPATH
 		"Full path to suppression file for valgrind")
 
+set (ENABLE_IO_ADAPTER_DBUS "ON"
```

Comment:
I am not very happy about even more CMake vars. Why not explictly exclude debian-musl at the place where needed? Btw. What is the compilation error, why cannot it be fixed?

Reply:
The [build job](https://build.libelektra.org/jenkins/job/elektra-gcc-configure-debian-musl/695/consoleFull) failed with `/usr/bin/ld: cannot find -ldbus-1`.
When configuring CMake it is present (`-- Found D-Bus: /usr/lib/x86_64-linux-gnu/libdbus-1.so`) so it seems like musl cannot link against it.
The dbus plugin was already disabled for this build job.

I agree. I will remove the variable and check with `${CMAKE_C_COMPILER} MATCHES "musl"`.

Reply:
We should be very precise here with the exclusion, otherwise we will disable these parts for OpenWRT, too.

Why does it even try to link against dbus? Only the plugin should link against dbus and the plugin is not built on our debian-musl, isn't it?

If the exclusion is really necessary, I am afraid we need to write a little test program with `try_compile` which tries to compile and link with dbus. This is already done several times, e.g.  in "src/plugins/augeas/CMakeLists.txt:              try_compile (HAS_LIBAUGEAS16"

Reply:
Okay, sure we don't want that.

Additional to the I/O bindings there are reusable "adapters" for D-Bus und ZeroMq (different PR). Plugins can use these adapters. 
It tries to link against D-Bus since these adapters are built as long as the required libraries are available - I didn't want to introduce a new build variable and these adapters are not intended to be used outside elektra.

`try_compile` seems more reasonable (testing capability vs. testing presumed capability) - I'll use that.

Reply:
The initial idea was that everything with deps should be either a tool, plugin or a binding. Then users can always manually exclude tools, plugins or bindings as wished or to make Elektra compile on erroneous platforms (which claim something is available but it isn't).

It seems like, however, that now also libs start to have external deps. (Yours is not the first, with src/libs/pluginprocess we already had similar problems.) So a LIBS (like BINDINGS and PLUGINS) might be necessary in the long run?

Reply:
It may be necessary, but for libs that are not only used internally I think we should prevent that. In my opinion users should not bother too much with internals.

For the adapter problem I could try to change it into an object library. The CMake documentation claims it does not link the files so the dbus issue would be moved to plugins using the adapters.